hip/solver/batch_cg_kernels.hip.cpp

```diff
@@ -91,17 +110,44 @@ class KernelCaller {
     {
         using real_type = gko::remove_complex<value_type>;
         const size_type nbatch = a.num_batch;
-
-        const int shared_size =
-            gko::kernels::batch_cg::local_memory_requirement<value_type>(
-                a.num_rows, b.num_rhs) +
-            PrecType::dynamic_work_size(a.num_rows, a.num_nnz) *
-                sizeof(value_type);
-
+        const int shared_gap = ((a.num_rows - 1) / 8 + 1) * 8;
```

Comment:
Inconsistent but correct. :)
```suggestion
        const int align_multiple = 8;
        const int shared_gap = ((a.num_rows - 1) / align_multiple + 1) * align_multiple;
```